File: 2924-find-champion-ii/2924-find-champion-ii.java
import java.util.*;

class Solution {
    public int findChampion(int n, int[][] edges) {
        boolean[] visited = new boolean[n];
        for (int[] edge : edges) {
            visited[edge[1]] = true;
        }
        
        int champion = -1;
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                if (champion == -1) {
                    champion = i;
                } else {
                    return -1;
                }
            }
        }
        
        return champion;
    }
}

File: 2924-find-champion-ii/2924-find-champion-ii.py
class Solution:
    def findChampion(self, n, edges):
        visited = [0] * n
        for u, v in edges:
            visited[v] = 1
        
        champion = -1
        for i in range(n):
            if visited[i] == 0:
                if champion == -1:
                    champion = i
                else:
                    return -1
        
        return champion
